### ISSUE TITLE
Actual Issue: Reserved Names should not allow to be added

### ISSUE THREAD

**Issue by brucevdkooij** [pods-framework/pods#5420]
**Describe the bug**
First time using Pods. Ran into this (reproduced using master): `classes/PodsAPI::save_pod` bails out before calling `cache_flush_pods` when it encounters any errors upon saving newly added fields. While it seems the other fields are still saved to the database, they are not visible to the end-user until the cache is flushed. 

Current revision line number [2356](https://github.com/pods-framework/pods/blob/28d6f0d51b6c4b1551372ed1cbe17c4338a5dcd4/classes/PodsAPI.php#L2356):

```
if ( ! empty( $errors ) ) {
    return pods_error( $errors, $this );
}

$this->cache_flush_pods( $pod );
```

(P.S. As far as I can tell there doesn't seem to be any error handling in place for failing to save fields.)

**To Reproduce**
1. Edit a pod
1. Add a field with name 'foobar'
1. Add a field with name 'name' (or 'post_title', or the name of an existing field)
1. Save pod

Observe: none of the newly addded fields are shown after saving

**Comment by jarrin** [comment]
+1 Having the same issue. Setting preserved names do not give an error.

**Comment by jimtrue** [comment]
See #4122 This should be applied to field names, taxonomy, post-types, etc. Someone should not be able to type ANY field name, post-type or taxonomy that is on the RESERVED Names Field List.

**Comment by raoulunger** [comment]
I filed a bug report  #(5440 )where I had the same issue with a field labeled 'status'. However, 'status' is not on Wordpress' Reserved Terms list. From the above mentioned case, 'name' is on that list, but 'foobar' is not. The same is true for the PHP list of reserved words.

Wordpress list: [(https://codex.wordpress.org/Reserved_Terms)](url)
PHP list [https://www.php.net/manual/en/reserved.php](url)

Perhaps some other list of reserved words is interefering too?

**Comment by JoryHogeveen** [comment]
@raoulunger Have you tried the latest PR concerning reserved keywords? #5441

**Comment by raoulunger** [comment]
Hi, not sure how to 'try a pr' - i'm not that deeply involved in coding, sorry...

**Comment by JoryHogeveen** [comment]
Yep, after investigation it turns out `status` is an alias for `post_status` within Pods (for use in Magic Tags etc.).
Full list of default keys and aliases:

```
ID                     ==> id
post_title             ==> title & name
post_content           ==> content
post_excerpt           ==> excerpt
post_author            ==> author
post_date              ==> created & date
post_date_gmt
post_status            ==> status
comment_status
ping_status
post_password
post_name              ==> slug &  permalink
to_ping
pinged
post_modified          ==> modified
post_modified_gmt
post_content_filtered
post_parent            ==> parent
guid
menu_order
post_type              ==> type
post_mime_type
comment_count
comments
```

@sc0ttkclark @jimtrue I think this issue can be closed, agreed?

**Comment by raoulunger** [comment]
Ok, that clarifies matters as far as 'status' is concerned — thanks for figuring this out, and for the list!

To solve the broader issue of using reserved word (this topic, and also #4122) it seems there would be a few options:

- code is changed so that aliases for id's can in fact be used, or (if that's not doable),
- a warning is shown when a 'forbidden' word is used, and
- some reference is made somewhere in the documentation about the list for no-go words

By the way, it's easy for me to suggest all this of course, as I can't really help here - my coding skills are far too limited, sorry!

Cheers!

**Comment by JoryHogeveen** [comment]
ID is simply a reserved keyword, I don't think we can change that.
About the warnings, there are currently shown but this is done through AJAX (server side) and not directly upon typing. I think this is something we will add in the future though but not as a maintenance release.
Documentation is always a good idea. @jimtrue Could you create a doc page for this that I can use?

**Comment by jimtrue** [comment]
Can't be closed until it's documented. Any other of those 'aliases' that would not work for creating as a field? ie 'title' 'parent' etc?

**Comment by JoryHogeveen** [comment]
This is all I could find so far (apart from the already mentioned WP core keywords)

**Comment by jimtrue** [comment]
This would still need to be patched because we have to explain WHY we're not allowing those fields to be added. Right now, we give them no information.

**Comment by JoryHogeveen** [comment]
That is not entirely true, AJAX returns an error message, but only upon saving.

**Comment by jimtrue** [comment]
I didn't receive any visible error when I duplicated the issue. If there's no error visible to the user, that's a problem.

**Comment by JoryHogeveen** [comment]
Did you test with the latest PR? #5443 
Keep in mind that these reserved keywords are allowed for fields names. Just not for Pods (CPT/Tax), see #5428.

**Comment by raoulunger** [comment]
I also did not receive an error message (therefor the confusion). I have not tested with the latest PR though, because I really have no clue how to do that (sorry again - i'm quite a newby here).

**Comment by JoryHogeveen** [comment]
@sc0ttkclark @jimtrue  Can this be closed?

**Comment by sc0ttkclark** [comment]
If we move the cache flushing before sending the pods_error, we can solve this ticket's main focus.

**Comment by JoryHogeveen** [comment]
Fixed in #5459

**Comment by raoulunger** [comment]
Sorry, late reaction, but thanks!
